open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/common/idempotency/IdempotencyValidator.java

```diff
@@ -81,26 +82,30 @@ public IdempotencyValidationResult validateIdempotency(ConsentManageData consent
         }
         try {
             String idempotencyKeyName = getIdempotencyAttributeName(consentManageData.getRequestPath());
-            // Retrieve consent ids that have the idempotency key name and value as attribute
-            ArrayList<String> consentIds = IdempotencyValidationUtils
-                    .getConsentIdsFromIdempotencyKey(idempotencyKeyName, idempotencyKeyValue);
-            // Check whether the consent id list is not empty. If idempotency key exists in the database then
-            // the consent Id list will be not empty.
-            if (!consentIds.isEmpty()) {
-                if (log.isDebugEnabled()) {
-                    log.debug(String.format("Idempotency Key  %s exists in the database. Hence this is an" +
-                            " idempotent request", idempotencyKeyValue));
-                }
-                for (String consentId : consentIds) {
-                    DetailedConsentResource consentRequest = consentCoreService.getDetailedConsent(consentId);
-                    if (consentRequest != null) {
-                        return validateIdempotencyConditions(consentManageData, consentRequest);
-                    } else {
-                        String errorMsg = String.format(IdempotencyConstants.ERROR_NO_CONSENT_DETAILS, consentId);
-                        log.error(errorMsg);
-                        throw new IdempotencyValidationException(errorMsg);
+            if (!IdempotencyConstants.EMPTY_OBJECT.equals(consentManageData.getPayload().toString())) {
+                // Retrieve consent ids that have the idempotency key name and value as attribute
+                ArrayList<String> consentIds = IdempotencyValidationUtils
+                        .getConsentIdsFromIdempotencyKey(idempotencyKeyName, idempotencyKeyValue);
+                // Check whether the consent id list is not empty. If idempotency key exists in the database then
+                // the consent Id list will be not empty.
+                if (!consentIds.isEmpty()) {
+                    if (log.isDebugEnabled()) {
+                        log.debug(String.format("Idempotency Key  %s exists in the database. Hence this is an" +
+                                " idempotent request", idempotencyKeyValue));
+                    }
+                    for (String consentId : consentIds) {
+                        DetailedConsentResource consentRequest = consentCoreService.getDetailedConsent(consentId);
```

Comment:
shall we rename the variable to consentResource?

Reply:
Fixed in https://github.com/wso2/financial-open-banking/pull/57/commits/b97b4cfcd61c0469858fb5d245d2d40707fada90